haystack/preview/components/reader.py

```diff
@@ -0,0 +1,151 @@
+from pathlib import Path
+from typing import List, Optional, Tuple, Union
+from haystack.preview import component, Document, Answer
+from haystack.preview.pipeline import Pipeline
+from haystack.lazy_imports import LazyImport
+
+with LazyImport(message="Run 'pip install farm-haystack[inference]'") as torch_and_transformers_import:
+    from transformers import AutoModelForQuestionAnswering
+    from tokenizers import Tokenizer, Encoding
+    import torch
+
+
+@component
+class ExtractiveReader:
+    def __init__(
+        self, reader: Union[Path, str], device: Optional[str] = None, max_seq_length: int = 384, top_k: int = 10
```

Comment:
Should there be a default reader?

Reply:
Also, is there a better parameter name? This follows the HF convention where it may either be a HF hub id or a path to a local folder. HF always calls this `model_name_or_path`.

Reply:
> I'd say yes, let's put a default value.
Do you know which one you'd like to put? I feel like this is trade-off between not frustrating users because they have to download 1.7GB (deberta-large) or because the model isn't working well (tinyroberta-squad2).

Reply:
Uh good question... Maybe a compromise, medium model? There is a "base" roberta, which is still quite large but I'd imagine most regular laptops can still handle it. If so I think that a good choice.

Reply:
Alright I think that's a good compromise. I also have no clue what hardware to expect, so smaller models look like a safer bet for now

Reply:
I wouldn't call the parameter just `model`. I would expect an actual `model` instance then, it's almost like a convention in data science/machine learning circles. `model_name_or_path` makes it clear that this is a string / path.